### ISSUE TITLE
Problem when generating iaxo-d0 readout. Segmentation fault.

### ISSUE THREAD

**Issue by jgalan** [rest-for-physics/detectorlib#1]
When generating for example the readout at `iaxo-readouts` a segmentation fault occurs during the readout generation.

```
root [0] TRestDetectorReadout *r = new TRestDetectorReadout("readouts_IAXOD0.rml");
Loading readout. This might take few seconds

 *** Break *** segmentation violation
[/usr/lib/system/libsystem_platform.dylib] _sigtramp (no debug info)
[<unknown binary>] (no debug info)
[/Users/javi/git/rest/install/lib/libRestDetector.dylib] TRestDetectorReadout::InitFromConfigFile() (no debug info)
[/Users/javi/git/rest/install/lib/libRestFramework.dylib] TRestMetadata::LoadConfigFromElement(TiXmlElement*, TiXmlElement*, std::__1::map<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::less<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > >, std::__1::allocator<std::__1::pair<std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > > > >) (no debug info)
[/Users/javi/git/rest/install/lib/libRestFramework.dylib] TRestMetadata::LoadConfigFromFile(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >, std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> >) (no debug info)
[/Users/javi/git/rest/install/lib/libRestDetector.dylib] TRestDetectorReadout::TRestDetectorReadout(char const*) (no debug info)
[/Users/javi/git/rest/install/lib/libRestDetector.dylib] TRestDetectorReadout::TRestDetectorReadout(char const*) (no debug info)
[<unknown binary>] (no debug info)
[<unknown binary>] (no debug info)
[/usr/local/lib/libCling.so] cling::IncrementalExecutor::runStaticInitializersOnce(cling::Transaction const&) const (no debug info)
[/usr/local/lib/libCling.so] cling::Interpreter::executeTransaction(cling::Transaction&) (no debug info)
[/usr/local/lib/libCling.so] cling::IncrementalParser::commitTransaction(llvm::PointerIntPair<cling::Transaction*, 2u, cling::IncrementalParser::EParseResult, llvm::PointerLikeTypeTraits<cling::Transaction*>, llvm::PointerIntPairInfo<cling::Transaction*, 2u, llvm::PointerLikeTypeTraits<cling::Transaction*> > >&, bool) (no debug info)
[/usr/local/lib/libCling.so] cling::IncrementalParser::Compile(llvm::StringRef, cling::CompilationOptions const&) (no debug info)
[/usr/local/lib/libCling.so] cling::Interpreter::EvaluateInternal(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, cling::CompilationOptions, cling::Value*, cling::Transaction**, unsigned long) (no debug info)
[/usr/local/lib/libCling.so] cling::Interpreter::process(std::__1::basic_string<char, std::__1::char_traits<char>, std::__1::allocator<char> > const&, cling::Value*, cling::Transaction**, bool) (no debug info)
[/usr/local/lib/libCling.so] cling::MetaProcessor::process(llvm::StringRef, cling::Interpreter::CompilationResult&, cling::Value*, bool) (no debug info)
[/usr/local/lib/libCling.so] HandleInterpreterException(cling::MetaProcessor*, char const*, cling::Interpreter::CompilationResult&, cling::Value*) (no debug info)
[/usr/local/lib/libCling.so] TCling::ProcessLine(char const*, TInterpreter::EErrorCode*) (no debug info)
[/usr/local/lib/libRint.so] TRint::ProcessLineNr(char const*, char const*, int*) (no debug info)
[/usr/local/lib/libRint.so] TRint::HandleTermInput() (no debug info)
[/usr/local/lib/libCore.so] TUnixSystem::CheckDescriptors() (no debug info)
[/usr/local/lib/libCore.so] TMacOSXSystem::DispatchOneEvent(bool) (no debug info)
[/usr/local/lib/libCore.so] TSystem::InnerLoop() (no debug info)
[/usr/local/lib/libCore.so] TSystem::Run() (no debug info)
[/usr/local/lib/libCore.so] TApplication::Run(bool) (no debug info)
[/usr/local/lib/libRint.so] TRint::Run(bool) (no debug info)
[/Users/javi/git/rest/install/bin/restRoot] main (no debug info)
[/usr/lib/system/libdyld.dylib] start (no debug info)
[<unknown binary>] (no debug info)
Root > .q
```

**Comment by jgalan** [comment]
I found the source of the error at `iaxo-readouts` project. The problem was that the RML was outdated. I have updated the RML at the `iaxo-readouts` project. See commit 5c18846

Now, I have added an exit code so that we bypass the seg.fault and we get some info about the problem.

```
root [0] TRestDetectorReadout *rr = new TRestDetectorReadout("readouts_IAXOD0.rml")
Loading readout. This might take few seconds
-- Error : TRestMetadata::ReplaceForLoopVars. Old for-loop construction identified
-- Error : Please, replace [] variable nomenclature by ${}.
```

There is still one problem with the readout generation. There are pixels outside the module boundaries definition. 

```
REST Warning (AddChannel) pixel outside the module boundaries
Channel: 237, Pixel : 0
Pixel origin = (-0.25 , 58.75)
Pixel size = (0.25 , 58.75)
Module size = (59.75 , 59.75)
REST Warning (AddChannel) pixel outside the module boundaries
Channel: 238, Pixel : 0
Pixel origin = (-0.25 , 59.25)
Pixel size = (0.25 , 59.25)
Module size = (59.75 , 59.75)
REST Warning (AddChannel) pixel outside the module boundaries
Channel: 239, Pixel : 0
Pixel origin = (-0.25 , 59.75)
Pixel size = (0.25 , 59.75)
Module size = (59.75 , 59.75)
```

Probably the best would be to follow the same approach as in `trexdm-readouts` where we define the module, `microbulkModule.rml` as an independent file to be used as reference in the main file. This needs a bit of review for the IAXO readouts. We should leave this issue open till this is fixed.

**Comment by cmargalejo** [comment]
I solved this and pushed it to GitLab https://lfna.unizar.es/iaxo/iaxo-readouts/-/commit/e620f9690abefee887df5ef0899035e8bb5f9f1f

Now I think the commit pointing to the iaxo-readouts submodule has to be updated.

**Comment by jgalan** [comment]
Awesome! I also updated the RML to v2.3 version. Just changing `TRestReadout` keyword by `TRestDetectorReadout`.

Also, in order to update the project at the main `framework` repository I tagged the first version of the projetct `v1.0`.

This allows to add a short info about v1.0 if you wish the update at the following page

https://lfna.unizar.es/iaxo/iaxo-readouts/-/tags

As you see, the latest commit just mentions `iaxo-readouts` to the version 1.0. 

https://github.com/rest-for-physics/framework/commits/development

One would go then to the specific project tagging to check what are the corresponding updates on the specific repository.

I let you close this issue.
